### PR TITLE
[feat] connection management with microsoft

### PR DESCRIPTION
**Pull request by lucas-debize**

### CODE REVIEW

File: frontend/components/NavBar.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import Link from '../../node_modules/next/link';
+import Link from '../node_modules/next/link';
```
Comment (thmarinho) [inline frontend/components/NavBar.tsx]: No imports from node_modules

File: frontend/components/msalConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export const msalConfig = {
+    auth: {
+        clientId: process.env.CLIENT_ID || 'ee231684-7664-48cf-9162-2468545eac5e',
+        authority: `${'https://login.microsoftonline.com/'}${'901cb4ca-b862-4029-9306-e5cd0f6d9f86'}`,
+        redirectUri: process.env.REDIRECT_URI || 'http://localhost:3001/auth/redirect',
```
Comment (thmarinho) [inline frontend/components/msalConfig.tsx]: Is this a prank ?

**Review by thmarinho — CHANGES_REQUESTED** [review]
Aren't you missing somethin like this ?
```jsx
      <AuthenticatedTemplate>
        <PageLayout />
      </AuthenticatedTemplate>

      <UnauthenticatedTemplate>
        <Login />
      </UnauthenticatedTemplate>
   ```

File: frontend/components/msalConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import * as dotenv from 'dotenv';
+dotenv.config({ path: '.env' });
+
+export const msalConfig = {
+    auth: {
+        clientId: process.env.NEXT_PUBLIC_CLIENT_ID || '',
+        authority: 'https://login.microsoftonline.com/' + (process.env.NEXT_PUBLIC_TENANT_ID) || '',
+        redirectUri: process.env.NEXT_PUBLIC_REDIRECT_URI || '',
```
Comment (thmarinho) [inline frontend/components/msalConfig.tsx]:
```suggestion
        clientId: process.env.NEXT_PUBLIC_CLIENT_ID,
        authority: 'https://login.microsoftonline.com/' + (process.env.NEXT_PUBLIC_TENANT_ID),
        redirectUri: process.env.NEXT_PUBLIC_REDIRECT_URI,
```

File: frontend/pages/_app.tsx [hunk 4]
```diff
@@ -1,23 +1,28 @@
...
-                pauseOnHover= {true}
-                draggable= {false}
-            />
-        </RootLayout>
+        <>
+            <MsalProvider instance={msalInstance}>
+                <AuthenticatedTemplate>
+                    <RootLayout>
+                        <Component {...pageProps} />
+                    </RootLayout>
```
Comment (thmarinho) [inline frontend/pages/_app.tsx]:
```suggestion
                    <RootLayout>
                        <Component {...pageProps} />
                                    <ToastContainer 
                                        position= 'top-right'
                                        autoClose= {5000}
                                        hideProgressBar= {false}
                                        closeOnClick= {true}
                                        pauseOnHover= {true}
                                        draggable= {false}
                                    />
                    </RootLayout>
```

**Review by thmarinho — CHANGES_REQUESTED** [review]

File: frontend/components/msalConfig.tsx (hunk 3 above)
Comment (lucas-debize) [reply frontend/components/msalConfig.tsx]: can't remove this without an error with msal definition

**Review by thmarinho — APPROVED** [review]

**Pull request merged**